lmdeploy/turbomind/deploy/target_model/base.py

```diff
@@ -83,6 +83,16 @@ def valid(self):
         return True
 
 
+_WEIGHT_DTYPE_MAP = dict(
+    w4=torch.float16,
```

Comment:
what's the difference between `w4` and `int4`?

Reply:
Here is the supported `weight_type` in turbomind. https://github.com/InternLM/lmdeploy/blob/af5a3edb49094e298731529a085866f9fd038be9/src/turbomind/triton_backend/llama/LlamaTritonModel.cc#L204

Reply:
should we map `fp32` to `torch.float16`?

Reply:
> what's the difference between `w4` and `int4`?

No difference, just in case
https://github.com/InternLM/lmdeploy/blob/af5a3edb49094e298731529a085866f9fd038be9/lmdeploy/turbomind/deploy/converter.py#L214-216



> should we map `fp32` to `torch.float16`?

This mapping is used to support param tensor cast. fp32 param would be cast to fp16 before exporting.

Reply:
> Here is the supported `weight_type` in turbomind.
> 
> https://github.com/InternLM/lmdeploy/blob/af5a3edb49094e298731529a085866f9fd038be9/src/turbomind/triton_backend/llama/LlamaTritonModel.cc#L204

Do we have int8 support now?

Reply:
int8 weight not yet

Reply:
If they are no difference, may use
```
_WEIGHT_DTYPE_MAP = dict(
   int4=torch.float16,
    fp16=torch.float16,
    fp32=torch.float16,
    bf16=torch.bfloat16 if torch.cuda.is_bf16_supported() else torch.float16,
)
    
```